### PR TITLE
started on custom_executor docs.

### PR DESCRIPTION
**Pull request by klemens-morgenstern**

### CODE REVIEW

File: doc/background/custom_executors.adoc [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  // prefer an executor with a new property. the executor may or may not support it.
+  never_blocking_executor prefer(const execution::blocking_t::never_t);
+  // not supported
+  example_executor prefer(const execution::blocking_t::always_t);
+};
+----
+
+==== Properties of the `asio::any_io_executor`
+
+In order to wrap an executor in an `asio::any_io_executor` two properties are required:
```
Comment (anarthal) [inline doc/background/custom_executors.adoc:54]: I think this is incomplete, looking at the source code:

      execution::context_as_t<execution_context&>,
      execution::blocking_t::never_t,
      execution::prefer_only<execution::blocking_t::possibly_t>,
      execution::prefer_only<execution::outstanding_work_t::tracked_t>,
      execution::prefer_only<execution::outstanding_work_t::untracked_t>,
      execution::prefer_only<execution::relationship_t::fork_t>,
      execution::prefer_only<execution::relationship_t::continuation_t>

Comment (klemens-morgenstern) [reply doc/background/custom_executors.adoc:54]: The `prefer_only`s are required?

**Comment by anarthal** [comment]
No. `prefer_only` makes the property on `any_io_executor` preferable, but not requirable.

This is a guess, but it's likely that the only hard requirement is `blocking.never` and the context one (that's used to retrieve the execution_context pointer).

My theory is that if a property cannot be required, prefer will just return the original executor. But I'd have to test it.

**Comment by klemens-morgenstern** [comment]
Tested it, seems to be the case.

**Comment by klemens-morgenstern** [comment]
See https://gcc.godbolt.org/z/qEj98M5KT

File: doc/background/custom_executors.adoc [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+which supports properties (see below) is equality comparable and has an `execute` function.
+
+==== `execute`
+
+[source,cpp]
+----
+struct example_executor
+{
+  template<typename Fn>
+  void executor(Fn && fn) const;
```
Comment (anarthal) [inline doc/background/custom_executors.adoc:22]: Shouldn't this be `execute`?

File: doc/background/custom_executors.adoc [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+  template<class F>
+  void execute(F f) const;
+
+  bool operator==(minimal_executor const &other) const noexcept;
+  bool operator!=(minimal_executor const &other) const noexcept;
+};
+----
+
+NOTE: See example/python.cpp for an implementation using python's `asyncio` event-loop.
```
Comment (anarthal) [inline doc/background/custom_executors.adoc:152]: A link to the file would be nice

File: doc/background/custom_executors.adoc (hunk 2 above)
Comment (klemens-morgenstern) [reply doc/background/custom_executors.adoc:22]: yes

File: doc/background/custom_executors.adoc [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  asio::execution_context &query(asio::execution::context_t) const;
+
+  static constexpr asio::execution::blocking_t
+  query(asio::execution::blocking_t) noexcept
+  {
+    return asio::execution::blocking.never;
+  }
+
+  template<class F>
+  void execute(F f) const;
```
[the remark this answers is not in the document]
Comment (klemens-morgenstern) [reply doc/background/custom_executors.adoc:145]: yes

**Pull request merged**